workbench-core/authorization/src/dynamicAuthorization/dynamoDBDynamicPermissionsPlugin.ts

```diff
@@ -0,0 +1,491 @@
+      identityId: groupId
+    });
+    const removeIdentityPermissionsPromises = _.chunk(identityPermissions, 25).map(
+      async (identityPermissionsChunk) => {
+        const deleteKeys = identityPermissionsChunk.map((identityPermission) => {
+          const item = this._createItemFromIdentityPermission(identityPermission);
+          return {
+            pk: item.pk,
+            sk: item.sk
+          };
+        });
+        await this._awsService.helpers.ddb
+          .batchEdit({
+            addDeleteRequests: deleteKeys
+          })
+          .execute();
+      }
+    );
+    await Promise.all(removeIdentityPermissionsPromises);
+    // Remove group entry
+    const groupKey = {
+      pk: this._composeIdentityKey(this._groupPrefix, 'GROUP', deleteGroupRequest.groupId),
+      sk: this._composeIdentityKey(this._groupPrefix, 'GROUP', deleteGroupRequest.groupId)
+    };
+
+    await this._awsService.helpers.ddb.delete(groupKey).execute();
+    return { deleted: true };
+  }
+
+  public async getUserGroups(getUserGroupsRequest: GetUserGroupsRequest): Promise<GetUserGroupsResponse> {
+    this._isInit();
+
+    const key = {
+      name: 'pk',
+      value: this._composeIdentityKey(this._assignedGroupPrefix, 'USER', getUserGroupsRequest.userId)
+    };
+
+    const groupIds: string[] = [];
+
+    const response: QueryCommandOutput = await this._awsService.helpers.ddb
+      .query({
+        key
+      })
+      .execute();
+
+    if (response && response.Items) {
+      response.Items.forEach((item) => {
+        const sk = _.get(item as { [key: string]: unknown }, 'sk') as string;
+        groupIds.push(this._decomposeKey(sk).id);
+      });
+    }
+    return {
+      groupIds
+    };
+  }
+
+  public async getUsersFromGroup(
+    getUsersFromGroup: GetUsersFromGroupRequest
+  ): Promise<GetUsersFromGroupResponse> {
+    //Requires a GSI where the partitionKey is sk and sort key is pk
+    this._isInit();
+    const key = {
+      name: 'sk',
+      value: this._composeIdentityKey(this._assignedGroupPrefix, 'GROUP', getUsersFromGroup.groupId)
+    };
+    const response = await this._awsService.helpers.ddb
+      .query({
+        index: this._getResourceBySortKeyIndex,
+        key
+      })
+      .execute();
+    const userIds: string[] = [];
+    if (response && response.Items) {
+      response.Items.forEach((item) => {
+        const pk = _.get(item as { [key: string]: unknown }, 'pk') as string;
+        userIds.push(this._decomposeKey(pk).id);
+      });
+    }
+    return {
+      userIds
+    };
+  }
+
+  public async createIdentityPermissions(
+    createIdentityPermissionsRequest: CreateIdentityPermissionsRequest
+  ): Promise<CreateIdentityPermissionsResponse> {
+    this._isInit();
+    const { identityPermissions } = createIdentityPermissionsRequest;
+
+    const unprocessedIdentityPermissions: IdentityPermission[] = [];
+    const batchCreateRequestPromises = identityPermissions.map(async (identityPermission) => {
```

Comment:
does this need `_.chunk(....`?

Reply:
No it doesn't need to be because it is using update and using node.js concurrency to run it instead of doing an actual batchWriteItem. This was meant to ensure we were tracking permissions that were not being properly processed